Allow sparse rhs to be input as SparseVector

Pbellive opened the pull request:
Currently need to input an n X 1 `SparseMatrixCSC` for single sparse rhs. Also added precompilation.

Sacha0 commented on src/MUMPSfuncs.jl:
```diff
@@ -116,6 +116,19 @@ function applyMUMPS!(factor::MUMPSfactorization{Float64},rhs::SparseMatrixCSC{Fl
+function applyMUMPS!(factor::MUMPSfactorization{Float64},rhs::SparseVector{Float64}, x::Array{Float64,1},tr::Int=0)
+	nrhs = 1
+	nzrhs = nnz(rhs)
+	colptr = [1;3]
```
Should this `colptr` match the `colptr` for a `SparseMatrixCSC` representation of `rhs::SparseVector`? If so, shouldn't `colptr = [1, nzrhs+1]`? If not, kindly ignore :).

Pbellive commented:
You're right of course @Sacha0. The lines you flagged should have been
```
colptr = [1; nzrhs + 1]
```
I pushed the wrong thing. Sorry about that and thanks for pointing it out. I fixed the error, added tests, and organized the tests in to sets. Think this should be good to go now.

Pbellive commented:
Is everyone ok to merge this now? I'm happy to make changes if needed but if people are happy with it as is, it would be great to get this merged. @lruthotto are you currently the only person with permission to merge PRs for this repo?

Thanks!

lruthotto commented:
Sorry for the wait, Patrick. This looks fabulous (as usual). I think everyone in JuliaSparse can merge this.
